src/Compilers/CSharp/Portable/Symbols/Retargeting/RetargetingNamedTypeSymbol.cs

```diff
@@ -477,5 +483,106 @@ internal override string? ExtensionGroupingName
 
         internal override string? ExtensionMarkerName
             => _underlyingType.ExtensionMarkerName;
+
+#nullable enable
+
+        internal ImmutableArray<(Cci.INestedTypeReference GroupingType, ImmutableArray<Cci.INestedTypeReference> MarkerTypes)> GetExtensionGroupingAndMarkerTypesForTypeForwarding(EmitContext context)
+        {
+            if (_lazyExtensionGroupingAndMarkerTypesForTypeForwarding.IsDefault)
```

Comment:
Is it guaranteed that this won't be called twice with a different `EmitContext` (which could mean getting a stale cached value)?